### PR TITLE
feat(refs T36187): errors admin user

### PR DESCRIPTION
**Pull request by elmasdemos**
**Ticket:** https://yaits.demos-deutschland.de/T36187

- fix list expandable
- fix deleting 
- fix updating orga and department

##How to review/test
Login support -> Nutzende -> expand user -> try deleting user and updating orga and department

### CODE REVIEW

File: client/js/components/user/DpUserListExtended.vue [hunk 1]
```diff
@@ -108,14 +108,12 @@ export default {
       ],
       isFiltered: false,
       isLoading: true,
-      organisations: []
+      organisations: [],
+      departments: []
```
Comment (spiess-demos) [inline client/js/components/user/DpUserListExtended.vue:112]: nitpick: alphabetic sorting

File: client/js/components/user/DpUserListExtended.vue [hunk 2]
```diff
@@ -151,10 +150,8 @@ export default {
         return
       }
 
+      dplan.notify.notify('confirm', Translator.trans('confirm.user.deleted'))
```
Comment (spiess-demos) [inline client/js/components/user/DpUserListExtended.vue:153]: Placing the sucess notification outside the `then()` will trigger it also if the operation wasn't successful.

File: client/js/components/user/DpUserListExtended.vue [hunk 3]
```diff
@@ -163,42 +160,40 @@ export default {
       }
 
       ids.map(id => {
+        dplan.notify.notify('confirm', Translator.trans('confirm.user.deleted'))
```
Comment (spiess-demos) [inline client/js/components/user/DpUserListExtended.vue:163]: Placing the sucess notification outside the `then()` will trigger it also if the operation wasn't successful.

File: client/js/components/user/DpUserListExtended.vue [hunk 4]
```diff
@@ -130,6 +128,7 @@ export default {
     },
 
     selectedItems () {
+      // The prop `itemSelections` and the method `dpToggleOne` are from `dpSelectAllMixin`
```
Comment (muellerdemos) [inline client/js/components/user/DpUserListExtended.vue:131]: thank you for this explanatory comment :+1:  this was not so easy to find out

File: client/js/components/user/DpUserListExtendedItem.vue [hunk 5]
```diff
@@ -273,16 +288,8 @@ export default {
...
-      if (typeof this.currentOrganisation.departments.find(dep => dep.id === this.currentDepartment.id) === 'undefined') {
-        this.resetCurrentDepartment()
-      }
-
       return dpApi.patch(url, {}, payload)
-        .then(response => checkResponse(response, {
-          200: { type: 'confirm', text: 'info.user.updated' },
-          204: { type: 'confirm', text: 'info.user.updated' }
-        }))
+        .then(checkResponse)
```
Comment (spiess-demos) [inline client/js/components/user/DpUserListExtendedItem.vue:292]: Why do you remove those custom notices?

File: client/js/components/user/DpUserListExtended.vue (hunk 1 above)
Comment (sakutademos) [reply client/js/components/user/DpUserListExtended.vue:112]: done here https://github.com/demos-europe/demosplan-core/pull/2673/commits/8a4018547e97324776689b21b7bdf28226bb7a81

File: client/js/components/user/DpUserListExtended.vue (hunk 2 above)
Comment (sakutademos) [reply client/js/components/user/DpUserListExtended.vue]: done here https://github.com/demos-europe/demosplan-core/pull/2673/commits/8a4018547e97324776689b21b7bdf28226bb7a81

File: client/js/components/user/DpUserListExtended.vue (hunk 3 above)
Comment (sakutademos) [reply client/js/components/user/DpUserListExtended.vue]: done here https://github.com/demos-europe/demosplan-core/pull/2673/commits/8a4018547e97324776689b21b7bdf28226bb7a81

File: client/js/components/user/DpUserListExtendedItem.vue (hunk 5 above)
Comment (sakutademos) [reply client/js/components/user/DpUserListExtendedItem.vue]: done here https://github.com/demos-europe/demosplan-core/pull/2673/commits/8a4018547e97324776689b21b7bdf28226bb7a81

**Review by muellerdemos — CHANGES_REQUESTED** [review]
There are still some problems when changing the organisation and/or the department within the userlist 
image

Right now I can not exactly say if this is a backend issue or if we send a invalid request via frontend. I can offer to investigate this together on next monday. 

```
#message: "The resource type `User` is not allowed for JSON:API `update` requests."
    #code: 0
    #file: "[/srv/www/demosplan/DemosPlanCoreBundle/Controller/GenericApiController.php](http://bobhh.dplan.local/app_dev.php/_profiler/open?file=demosplan/DemosPlanCoreBundle/Controller/GenericApiController.php&line=128#line128)"
```

**Comment by spiess-demos** [comment]
- the patch request is only allowed [for users with both permissions](https://github.com/demos-europe/demosplan-core/blob/b24046925e8c3619e107252db2a80e5d3a56ad42/demosplan/DemosPlanCoreBundle/ResourceTypes/UserResourceType.php#L78-L82)
- `feature_user_edit` is [disabled for support users ](https://github.com/demos-europe/demosplan-project-bobhh/blob/f3a08a37196c116670781340e8dada67d75fcdd6/app/Resources/DemosPlanCoreBundle/Logic/Permissions.php#L141)in certain projects
- simply enabling it will not suffice as the "classic" user list is shown in that case, due to insufficient permission design - the choice of which userlist to show [is coupled to a specific combination of crud permissions](https://github.com/demos-europe/demosplan-core/blob/06d2b39d29e78354de2c5f85068adea1f13b9623/templates/bundles/DemosPlanCoreBundle/DemosPlanUser/list_user.html.twig#L36-L45), while we would need a new permission like `feature_userlist_extended`

File: client/js/components/user/DpUserListExtendedItem.vue (hunk 5 above)
Comment (gruenbergerdemos) [reply client/js/components/user/DpUserListExtendedItem.vue]: I fixed the checkResponse and forgot to add the custom notifications. Thanks for adding.

File: config/permissions.yml [hunk 6]
```diff
@@ -2319,6 +2319,12 @@ feature_user_list:
     expose: true
     label: 'fetch users from the API route.'
     loginRequired: true
+# Only needed for user list in one specific project
+feature_user_list_extended:
```
Comment (jalalifardemos) [inline config/permissions.yml:2323]: should be add in bobhh/permissions.php under support role, If you add this permission, please add the link of bobhh branch in Description->linked branches

**Review by jalalifardemos — CHANGES_REQUESTED** [review]

File: config/permissions.yml (hunk 6 above)
Comment (gruenbergerdemos) [reply config/permissions.yml:2323]: Yes, I added it. You see the link below.

**Review by jalalifardemos — APPROVED** [review]
Lgtm, code review only. Thanks

**Pull request merged**